src/modules/Game.class.js

```diff
@@ -50,17 +184,81 @@ class Game {
    * `win` - the game is won;
    * `lose` - the game is lost
    */
-  getStatus() {}
+  getStatus() {
+    if (this.status === 'idle') {
+      return 'idle';
+    }
+
+    if (this.checkWin()) {
+      return 'win';
+    }
+
+    if (this.checkLose()) {
+      return 'lose';
+    }
+
+    return 'playing';
+  }
+
+  checkWin() {
+    return this.board.some((row) => row.includes(2048));
+  }
+
+  checkLose() {
+    if (this.board.some((row) => row.includes(0))) {
+      return false;
+    }
+
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 4; j++) {
+        if (
+          (j < 3 && this.board[i][j] === this.board[i][j + 1]) ||
+          (i < 3 && this.board[i][j] === this.board[i + 1][j])
+        ) {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  }
 
   /**
    * Starts the game.
    */
-  start() {}
+  start() {
+    this.board = JSON.parse(JSON.stringify(this.defaultState));
+    this.score = 0;
+    this.status = 'playing';
+    this.addRandomTile();
+    this.addRandomTile();
+  }
+
+  addRandomTile() {
+    const emptyCells = [];
+
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 4; j++) {
+        if (this.board[i][j] === 0) {
+          emptyCells.push({ row: i, col: j });
+        }
+      }
+    }
+
+    if (emptyCells.length > 0) {
+      const { row, col } =
+        emptyCells[Math.floor(Math.random() * emptyCells.length)];
+
+      this.board[row][col] = Math.random() < 0.9 ? 2 : 4;
+    }
+  }
 
   /**
    * Resets the game.
    */
-  restart() {}
+  restart() {
```

Comment:
after clicking restart, the button text is "Start", but new tiles appeared and I can play